src/components/connect.js

```diff
@@ -179,7 +179,7 @@ export default function connect(mapPropsToRequestsToProps, options = {}) {
         const onRejection = this.createPromiseStateOnRejection(prop, mapping, startedAt)
 
         if (mapping.value) {
-          const meta = {}
+          const meta = mapping.meta || {}
           this.setAtomicState(prop, startedAt, mapping, initPS(meta))
```

Comment:
I went ahead and made this set the provided `meta` on the pending `PromiseState`; however, this is different behavior from the `value`, which is not set until the `PromiseState` is fulfilled. I'm not super happy about this, but I think it is the most expected behavior. We can think about setting `value` up front, but then it's different from regular URL-based requests. Feedback welcome.

cc: @jsullivan @hburrows